File: python/1535. Find the Winner of an Array Game.py
class Solution:
    def getWinner(self, arr: List[int], k: int) -> int:
        l = arr[0]
        winStreak = 0
        for i in range(1,len(arr)):
            r = arr[i]
            if l > r:
                winStreak += 1
            else:
                winStreak = 1
                l = r
            if winStreak == k:
                return l
        return l

File: java/1535. Find the Winner of an Array Game.java
class Solution {
    public int getWinner(int[] arr, int k) {
        int left = 0;
        int right = 1;
        int winCount = 0;
        while(right < arr.length){
            int l = arr[left];
            int r = arr[right];
            if( l > r){
                winCount+=1;
                right += 1;
            }
            else{
                winCount = 1;
                left = right;
                right = left + 1;
            }
            if(winCount == k){
                return arr[left];
            }
        }
        return arr[left];
    }
}